cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/batch/AppWithLocalFiles.java

```diff
@@ -0,0 +1,137 @@
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.io.IntWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.Reducer;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+/**
+ * App to test local files in programs.
+ */
+public class AppWithLocalFiles extends AbstractApplication {
+  public static final String MR_INPUT_DATASET = "input";
+  public static final String MR_OUTPUT_DATASET = "output";
+  public static final String STOPWORDS_FILE_ARG = "stopwords.file";
+
+  @Override
+  public void configure() {
+    createDataset(MR_INPUT_DATASET, KeyValueTable.class);
+    createDataset(MR_OUTPUT_DATASET, KeyValueTable.class);
+    addStream("LocalFileStream");
+    addMapReduce(new MapReduceWithLocalFiles());
+  }
+
+  public static class MapReduceWithLocalFiles extends AbstractMapReduce {
+
+    @Override
+    public void beforeSubmit(MapReduceContext context) throws Exception {
+      Map<String, String> args = context.getRuntimeArguments();
+      if (args.containsKey(STOPWORDS_FILE_ARG)) {
+        context.addLocalFile(URI.create(args.get(STOPWORDS_FILE_ARG)));
+      }
+      context.setInput(args.get(MR_INPUT_DATASET));
+      context.addOutput(args.get(MR_OUTPUT_DATASET));
+      Job job = context.getHadoopJob();
+      job.setMapperClass(TokenizerMapper.class);
+      job.setReducerClass(IntSumReducer.class);
+    }
+
+    public static class TokenizerMapper extends Mapper<byte[], byte[], Text, IntWritable>
+      implements ProgramLifecycle<MapReduceContext> {
+
+      private static final IntWritable ONE = new IntWritable(1);
+      private Text word = new Text();
+      private final List<String> stopWords = new ArrayList<>();
+
+      @Override
+      public void map(byte[] key, byte[] value, Context context) throws IOException, InterruptedException {
+        StringTokenizer itr = new StringTokenizer(Bytes.toString(value));
+
+        while (itr.hasMoreTokens()) {
+          String token = itr.nextToken();
+          if (!stopWords.contains(token)) {
+            word.set(token);
+            context.write(word, ONE);
+          }
+        }
+      }
+
+      @Override
+      public void initialize(MapReduceContext context) throws Exception {
+        List<URI> localFiles = context.getLocalFiles();
+        Map<String, String> args = context.getRuntimeArguments();
+        if (args.containsKey(STOPWORDS_FILE_ARG)) {
+          String localFilePath = args.get(STOPWORDS_FILE_ARG);
+          for (URI localFile : localFiles) {
+            if (localFilePath.equals(localFile.toString())) {
+              try (FileSystem fs = FileSystem.get(localFile, new Configuration());
```

Comment:
So, I used it because I'm passing the I couldn't get ``java.io.File`` to find a file of the format ``file:/path/to/file``, which is what I get here from https://github.com/caskdata/cdap/pull/4414/files#diff-0642c8ed9095b2017093f641afd4fc7bR572